CMakeLists.txt

```diff
@@ -231,42 +231,34 @@ install(TARGETS ${OpenBLAS_LIBNAME}
 
 # Install include files
 	set (GENCONFIG_BIN ${CMAKE_BINARY_DIR}/gen_config_h${CMAKE_EXECUTABLE_SUFFIX})
-	ADD_CUSTOM_COMMAND(
-	OUTPUT ${CMAKE_BINARY_DIR}/openblas_config.h
-	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/config.h
-	COMMAND ${GENCONFIG_BIN} ${CMAKE_CURRENT_SOURCE_DIR}/config.h ${CMAKE_CURRENT_SOURCE_DIR}/openblas_config_template.h > ${CMAKE_BINARY_DIR}/openblas_config.h
-	)
 
-	ADD_CUSTOM_TARGET(genconfig
-	ALL
-	DEPENDS openblas_config.h
-	)
-	add_dependencies(genconfig ${OpenBLAS_LIBNAME})
+    execute_process(COMMAND ${GENCONFIG_BIN}
+                            ${CMAKE_CURRENT_SOURCE_DIR}/config.h
+                            ${CMAKE_CURRENT_SOURCE_DIR}/openblas_config_template.h
+                    OUTPUT_VARIABLE OPENBLAS_CONFIG_H_CONTENTS)
+
+    file(WRITE ${CMAKE_BINARY_DIR}/openblas_config.tmp "${OPENBLAS_CONFIG_H_CONTENTS}")
+    configure_file(${CMAKE_BINARY_DIR}/openblas_config.tmp ${CMAKE_BINARY_DIR}/openblas_config.h COPYONLY)
 
 	install (FILES ${CMAKE_BINARY_DIR}/openblas_config.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 	message(STATUS "Generating f77blas.h in ${CMAKE_INSTALL_INCLUDEDIR}")
 
-	ADD_CUSTOM_TARGET(genf77blas
-	ALL
-	COMMAND ${AWK} 'BEGIN{print \"\#ifndef OPENBLAS_F77BLAS_H\" \; print \"\#define OPENBLAS_F77BLAS_H\" \; print \"\#include \\"openblas_config.h\\" \"}; NF {print}; END{print \"\#endif\"}' ${CMAKE_CURRENT_SOURCE_DIR}/common_interface.h > ${CMAKE_BINARY_DIR}/f77blas.h
-	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/config.h
-	)
-	add_dependencies(genf77blas ${OpenBLAS_LIBNAME})
-
+    file(WRITE ${CMAKE_BINARY_DIR}/f77blas.tmp "")
+    file(APPEND ${CMAKE_BINARY_DIR}/f77blas.tmp "#ifndef OPENBLAS_F77BLAS_H\n#define OPENBLAS_F77BLAS_H\n#include \"openblas_config.h\"\n")
+    file(READ ${CMAKE_CURRENT_SOURCE_DIR}/common_interface.h COMMON_INTERFACE_H_CONTENTS)
+    file(APPEND ${CMAKE_BINARY_DIR}/f77blas.tmp "${COMMON_INTERFACE_H_CONTENTS}")
+    file(APPEND ${CMAKE_BINARY_DIR}/f77blas.tmp "#endif")
+    configure_file(${CMAKE_BINARY_DIR}/f77blas.tmp ${CMAKE_BINARY_DIR}/f77blas.h COPYONLY)
 	install (FILES ${CMAKE_BINARY_DIR}/f77blas.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 if(NOT NO_CBLAS)
 	message (STATUS "Generating cblas.h in ${CMAKE_INSTALL_INCLUDEDIR}")
 
-	ADD_CUSTOM_TARGET(gencblas
-	ALL
-	COMMAND ${SED} 's/common/openblas_config/g' ${CMAKE_CURRENT_SOURCE_DIR}/cblas.h > "${CMAKE_BINARY_DIR}/cblas.tmp"
-	COMMAND cp "${CMAKE_BINARY_DIR}/cblas.tmp" "${CMAKE_BINARY_DIR}/cblas.h"
-	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/cblas.h
-	)
-	add_dependencies(gencblas ${OpenBLAS_LIBNAME})
```

Comment:
Not sure I understand the question, but cblas.h is generated from openblas_config.h so needs to correspond to that at the time the library is to be installed. The sequence of actions was reordered by @nelish-msft recently, I think to avoid some circular dependency in Windows10 builds. Looking at the history of the CMakeLists file may tell if it was me who made a bad decision originally when I rewrote the header installation step (which used to dump all the internal headers into the install directory).